ui/dashboards/src/components/Panel/PanelActions.tsx

```diff
@@ -87,21 +95,53 @@ export const PanelActions: React.FC<PanelActionsProps> = ({
     }
     return undefined;
   }, [descriptionTooltipId, description]);
+
   const linksAction = links && links.length > 0 && <PanelLinks links={links} />;
   const extraActions = editHandlers === undefined && extra;
 
-  const queryStateIndicator = useMemo(() => {
-    const hasData = queryResults.some((q) => q.data);
-    const isFetching = queryResults.some((q) => q.isFetching);
+  const queryStateIndicator = useMemo((): ReactNode | undefined => {
+    // Check if any query is fetching
+    const isQueryFetching = queryResults.some((q) => q.isFetching);
+    // Get query-specific errors
     const queryErrors = queryResults.filter((q) => q.error);
-    if (isFetching && hasData) {
+
+    // Convert QueryData[] to TimeSeriesData format for data availability check
+    const timeSeriesData: TimeSeriesData | undefined =
```

Comment:
As stated before we don't need a specific format for the data as plugins contain more than time series data. I don't think we need to change this part of the code, as the logic should be present on the plugin